merge.c

```diff
@@ -3,7 +3,79 @@
 
 #include "merge.h"
 
-PyObject* merge( PyObject* self, PyObject* args )
+/* Heterogeneous compare: default, always safe to fall back on. */
+static inline int safe_object_compare(PyObject *v, PyObject *w)
```

Comment:
I have no idea if it'll compile, but it might be able to just define this as `static int safe_object_compare(PyObject *v, PyObject *w);` which will force the compile to look for the function in the python library. (or it might not...)